### PR TITLE
Ports Aurora's floor tile color turf decals (For mappers)

### PR DESCRIPTION
**Pull request by deathride58**
Title. This also puts turf decals in files separate from decals.dm, adds a shitton of colors to colors.dm in _DEFINES for future use, all copy-pasted from Aurora, and allows turf decals to pass their color to the decal component.

In layman's terms, this PR allows mappers to do shit like this without touching a single line of code
image
image
image

:cl: deathride58
add: Mappers can now color individual tiles of floors via turf decals.
/:cl:

These decals are a *lot* more flexible compared to the current method of coding in new tile color combinations as new turf subtypes. This also makes it much easier for mappers to add in new colors for existing turf decals.

### CODE REVIEW

**Comment by optimumtact** [comment]
hey, that's pretty good

**Comment by praisenarsie** [comment]
@deathride58 and you didn't even bother adding the floor painter. shame on you.

**Comment by ChangelingRain** [comment]
These seem much less textured than the current floor tiles, but that's at least easy to fix if desired since it's so little sprites to edit in comparison.

**Comment by epochayur** [comment]
this is seriously cool

**Comment by praisenarsie** [comment]
joan is right actually, they do look different. i'll show how tomorrow unless someone spots the Cool Mistake!

**Comment by deathride58** [comment]
Aye, these sprites are lifted directly from Aurora.

The only real noticeable issue is that only the bottom right tile has the subtle smudges present in all floor tiles, while the rest of the tiles lack those smudges. It should be a relatively easy, though tedious fix. That issue is pretty hard to notice ingame most of the time, though.

**Comment by coiax** [comment]
If the floor is broken and then replaced with a fresh tile, will the decals still be there

**Comment by deathride58** [comment]
@coiax No, unfortunately, but that's an issue with all turf decals.

**Comment by Mark9013100** [comment]
Could this pave the way for a floor painter object?

**Comment by deathride58** [comment]
@Mark9013100 yep! This PR makes it a lot easier to add that.

**Comment by ShizCalev** [comment]
https://imgur.com/a/ljaq4

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/obj/effect/turf_decal/corner
```
Comment (ShizCalev) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm]: Change the path to `/obj/effect/turf_decal/tile` instead

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+/obj/effect/turf_decal/corner/black
+	name = "black corner"
+	color = COLOR_ALMOST_BLACK
+
+/obj/effect/turf_decal/corner/black/diagonal
+	icon_state = "corner_white_diagonal"
+
+/obj/effect/turf_decal/corner/black/full
+	icon_state = "corner_white_full"
```
Comment (ShizCalev) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm]: These aren't actually full and should be part of the corner sprite instead as the intermediate directions.

**Review by ShizCalev — CHANGES_REQUESTED** [review]
Move the current intermediate directions of the corner_white sprite into a corner_white_side one, move corner_white_full's sprites into the intermediates of corner_white, and make the corner_white_full sprite contain 4 squares to color the entire tile instead.

**Comment by nicbn** [comment]
Ok so who's going to replace all the old tiles with this

**Comment by praisenarsie** [comment]
image
wow im a day late on showing the mistake, as you see here there's lack of shading/detail on some of the tiles.
compare the default floors to the colored one, there's (for some reason) only shading/detail on the bottom right tile of the colored floors

so fix it somehow 🥘

**Comment by deathride58** [comment]
@praisenarsie Fixed.

@nicbn Probably not me, replacing every single colored tile with these would be a pain in the ass.

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 3]
```diff
@@ -0,0 +1,197 @@
+/obj/effect/turf_decal/tile
+	icon_state = "corner_white_full"
```
Comment (ShizCalev) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm]: name = "tile decal"

File: code/game/objects/effects/decals/turfdecal/tilecoloring.dm [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+/obj/effect/turf_decal/tile/black
+	name = "black corner"
```
Comment (ShizCalev) [inline code/game/objects/effects/decals/turfdecal/tilecoloring.dm]: name = "black tile decal"

and the rest leading down.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by praisenarsie** [comment]
now add the fookin' floor painter

**Comment by ninjanomnom** [comment]
I'm so proud of my decal component. It's grown up, left the nest, then come back to visit me already T.T
